### PR TITLE
Eliminate unecessary copies with astype.

### PR DESCRIPTION
**Pull request by h-mayorquin**
https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html

The astype function in numpy creates copies by default even if not necessary. This PR eliminates some in the library as we slowly move to use more the proper pre-processing:

https://github.com/h-mayorquin/spikeinterface/blob/38ce4b5c7519c376e24de983889ba694e31ee84e/src/spikeinterface/preprocessing/astype.py#L6-L34

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Damned quality metrics.

**Comment by alejoe91** [comment]
Tests are undeterministic and potentially wrong..just comment it out :)

https://github.com/SpikeInterface/spikeinterface/pull/2221/files#diff-c96c047577a529810e0c49ea34471dcf5efd1e4699c0465dc58eb7cc52b7227fR396

**Comment by h-mayorquin** [comment]
@alejoe91 
You shame me by being kind when I am whining obnoxiously about parts of the library that I don't understand well! : D

In a more serious tone, I think that we can make this test pass by just supressing the type.

File: src/spikeinterface/curation/curation_tools.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def _find_duplicated_spikes_random(spike_train: np.ndarray, censored_period: int
 
 if HAVE_NUMBA:
 
-    @numba.jit((numba.int64[::1], numba.int32), nopython=True, nogil=True, cache=True)
```
Comment (h-mayorquin) [inline src/spikeinterface/curation/curation_tools.py:62]: I have no idea why this is failing:

image

But also last time I checked this did not make any difference. Maybe @DradeAW wants to fix this as this was important to him.

Feel free to push this or we can do this in another PR.

Comment (DradeAW) [reply src/spikeinterface/curation/curation_tools.py:62]: The `nopython=True, nogil=True` is important.
The other stuff is just there to reassure me ^^. I'm fine with removing them

Comment (h-mayorquin) [reply src/spikeinterface/curation/curation_tools.py:62]: @alejoe91 this should be ready to merge.

**Review by alejoe91 — APPROVED** [review]

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**